src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -427,23 +445,30 @@ private void determineTerritoriesToAttack(final Map<Territory, ProAttackTerritor
     ProLogger.debug("Final number of territories: " + (numToAttack - 1));
   }
 
-  private void determineTerritoriesThatCanBeHeld(final List<ProAttackTerritoryData> prioritizedTerritories,
-      final Map<Territory, ProAttackTerritoryData> attackMap,
-      final Map<Territory, ProAttackTerritoryData> enemyAttackMap, final Map<Territory, Double> territoryValueMap) {
+  private void determineTerritoriesThatCanBeHeld(final List<ProTerritory> prioritizedTerritories,
+      final Map<Territory, ProTerritory> attackMap, final ProMoveOptions enemyAttackOptions,
+      final Map<Territory, Double> territoryValueMap) {
 
     ProLogger.info("Check if we should try to hold attack territories");
 
     // Determine which territories to try and hold
     final Map<Unit, Territory> unitTerritoryMap = utils.createUnitTerritoryMap(player);
-    for (final ProAttackTerritoryData patd : prioritizedTerritories) {
+    for (final ProTerritory patd : prioritizedTerritories) {
       final Territory t = patd.getTerritory();
 
+      // If strafing then can't hold
+      if (patd.isStrafing()) {
```

Comment:
unblocking might be a better verb than strafing. For me strafing is when you attack with excessive firepower and HP and retreat after inflicting more damage, but don't want to capture the territory. For example, 9 tanks and 3 infantry attacking 6 infantry. If things go right, after one round  9 tanks and 1 infantry will retreat after having traded 2 infantry for 5.